src/EmbeddedTypefaceFinder.cpp

```diff
@@ -1,82 +1,97 @@
-#include "EmbeddedFontFinder.h"
-#include "EmbeddedFontFinder_internal.h"
+#include "EmbeddedTypefaceFinder.h"
+#include "EmbeddedTypefaceFinder_internal.h"
 
 #include "QtExtensions.h"
 #include "QtCommon.h"
 
 #include <SkTypeface.h>
-#include <SkPaint.h>
 
 #include "ICoreResourcesProvider.h"
 #include "SkiaUtilities.h"
 #include "Logging.h"
 
-OsmAnd::EmbeddedFontFinder::EmbeddedFontFinder(
+OsmAnd::EmbeddedTypefaceFinder::EmbeddedTypefaceFinder(
     const std::shared_ptr<const ICoreResourcesProvider>& coreResourcesProvider_ /*= getCoreResourcesProvider()*/)
     : coreResourcesProvider(coreResourcesProvider_)
 {
-    for (const auto& embeddedFontResource : constOf(resources))
+    for (const auto& embeddedTypefaceResource : constOf(resources))
     {
-        const auto fontData = coreResourcesProvider->getResource(embeddedFontResource);
-        if (fontData.isNull())
+        const auto typefaceData = coreResourcesProvider->getResource(embeddedTypefaceResource);
+        if (typefaceData.isNull())
         {
             LogPrintf(LogSeverityLevel::Error,
-                "Failed to load embedded font data for '%s'",
-                qPrintable(embeddedFontResource));
+                "Failed to load embedded typeface data for '%s'",
+                qPrintable(embeddedTypefaceResource));
             continue;
         }
 
-        const auto font = SkiaUtilities::createTypefaceFromData(fontData);
-        if (!font)
+        const auto skTypeface = SkiaUtilities::createTypefaceFromData(typefaceData);
+        if (!skTypeface)
         {
             LogPrintf(LogSeverityLevel::Error,
-                "Failed to create SkTypeface from embedded font data for '%s'",
-                qPrintable(embeddedFontResource));
+                "Failed to create SkTypeface from embedded data for '%s'",
+                qPrintable(embeddedTypefaceResource));
             continue;
         }
 
-        _fonts.push_back(font);
+        const auto hbBlob = std::shared_ptr<hb_blob_t>(
+            hb_blob_create_or_fail(
+                typefaceData.constData(),
+                typefaceData.length(),
+                HB_MEMORY_MODE_READONLY,
+                new QByteArray(typefaceData),
```

Comment:
why do you need to do that?

Reply:
That is a very valid question: in this case since I'm referencing directly data from `typefaceData` (which is `typefaceData`), I need to ensure that it survives as long as `hbBlob` exists. Technically, since `typefaceData` is in the same scope, it's not needed, yet to avoid being error-prone, does not hurt. `new QByteArray(typefaceData)` in this case creates a clone of `QByteArray` by incrementing reference to it's internal data structure - w/o any actual memory copy - see https://doc.qt.io/qt-5/qbytearray.html#QByteArray where it says that:

> This operation takes constant time, because QByteArray is implicitly shared. This makes returning a QByteArray from a function very fast. If a shared instance is modified, it will be copied (copy-on-write), taking linear time.

Since we're using the memory only in read-only mode, no copy-on-write will ever happen.